Issue multiple links + spotify link added (social media)

Kuriv2001 opened the pull request:
Through this PR, multiple possible patterns for group links can be added to a social media. This ensures, that no fake website can be given for people to be scammed, while still allowing multiple patterns of links for a group chat to be accepted. => This solves the problem mentioned in #206 

It also adds the possibility to share both Spotify playlists, as well as accounts, on top of the already existing social media groups, as was requested during the last sprint planning.

The fields for adding links at the creation of the event or while editing the event, also have been adjusted to be only one line big, like the other ones in our app, so no unintended action happens.

I also struggled quite a lot with making the test work with the newly added spotify links. I tried everything I could for multiple hours, like that the new component wasn't visible with the keyboard extended. I therefore implemented the ChooseSocialsScreen, such that on enter of the keyboard, the keyboard retracts. But nothing worked, so I decided to leave it, as it works fine locally and manually => issue will be created.

leagrieder reviewed (changes requested):
Good PR. Good size and good job correcting this security issue.

Almost everything looks good to me, however there is one thing that I find strange.. If I enter invalid links or other things in the social event rectangles I am allowed to modify/create the event. I think it might make more sense to not allow event creation if what is written somewhere is not correct right? I believe that even though it the event seems to be saved with those bad modifications you have correctly coded things such that it doesn't actually save the new invalid social media links but I still feel like it would be more user friendly to add a toast message if something that is entered is invalid. Let me know what you think.

Kuriv2001 commented:
> Good PR. Good size and good job correcting this security issue.
> 
> Almost everything looks good to me, however there is one thing that I find strange.. If I enter invalid links or other things in the social event rectangles I am allowed to modify/create the event. I think it might make more sense to not allow event creation if what is written somewhere is not correct right? I believe that even though it the event seems to be saved with those bad modifications you have correctly coded things such that it doesn't actually save the new invalid social media links but I still feel like it would be more user friendly to add a toast message if something that is entered is invalid. Let me know what you think.

Hey, Thank you for your thoughts. It does make a lot of sense and i changed everything now, such that you can't create the event if the social media links are not correct.

I also added all the logic to editEventScreen (also for title and dates) as this has not yet been thought of

leagrieder commented:
Looks good now!! I approve this

Athanaze commented on app/src/main/java/com/monkeyteam/chimpagne/ui/event/ChooseSocialsPanel.kt:
```diff
@@ -25,20 +42,23 @@ fun ChooseSocialsPanel(eventViewModel: EventViewModel) {
+              modifier = Modifier.testTag("social_media_title"))
           Spacer(modifier = Modifier.height(16.dp))
         }
+        for ((platform) in socialMediaStates) {
```
useless () around platform
